lib/join/iamjoin/iam.go

```diff
@@ -255,12 +261,31 @@ func checkIAMAllowRules(identity *AWSIdentity, allowRules []*types.TokenRule) er
 				continue
 			}
 		}
+
+		if len(rule.AWSOrganizationID) > 0 {
+			organizationID, err := organizationIDFetcher()
+			if err != nil {
+				return trace.Wrap(err)
+			}
```

Comment:
If the join rule asked for organization but none are available due to fetch error we should highlight this. This will be a common problem for sure. In fact, we should verify this before we start the actual RBAC check. Throwing an error here conflates "access denied" errors with "we don't have permissions to check the access properly".